### DISCUSSION TITLE
Clipboard startup cost

### DISCUSSION THREAD

**Discussion by adrian5** [LazyVim/LazyVim#147] [General]
Might not apply to all users, but something to consider documenting (requires individual configuration). I don't use LazyVim right now but on my system this line saves about 50ms (!) startup time:

```lua
vim.g.clipboard = { name = "xsel", cache_enabled = true }
                            ↑ clipboard provider
```

[neovim #9570](https://github.com/neovim/neovim/issues/9570)

**Comment by adrian5** [comment]
Hmm, actually the delay disappears when switching from `xsel` to the `xclip` utility (without having to set `vim.g.clipboard`). So maybe the issue is too niche and unlikely to affect other users on the major platforms.

Edit: Don't know what to think about [this](https://github.com/neovim/neovim/issues/20862#issuecomment-1296192535), hopefully the delay for `xsel` is fixed in 0.9 then.

**Reply by folke** [reply]
ok, cool. Seems to be fixed on 0.9 indeed.
I'm on wayland, so don't have that issue wih `wl-copy`

**Reply by thyttan** [reply]
I had this problem on Fedora 37, gnome terminal. Fixed by installing `wl-clipboard` where `wl-copy` lives. Thanks!
